src/pytest_benchmark/plugin.py

```diff
@@ -857,7 +866,10 @@ def display_results_table(self, tr):
                 worst[prop] = max(benchmark[prop] for _, benchmark in report_progress(
                     benchmarks, tr, "{line} ({pos}/{total})", line=line))
 
-            unit, adjustment = time_unit(best.get(self.sort, benchmarks[0][self.sort]))
+            time_unit_key = self.sort
+            if self.sort in ("name", "fullname"):
```

Comment:
Another option would mean `--benchmark-time-unit` it it could take values like: s, ms, us, ns, second, micro-second, microsecond, ... or min, max, avg, mean etc for automatic.

This don't need to be changed here, only asking for opinion.